src/discord/actions/approveChallenge.ts

```diff
@@ -0,0 +1,58 @@
+import { ButtonInteraction, CommandInteraction } from 'discord.js';
+
+import * as challenges from '../../challenges';
+import * as raffles from '../../raffles';
+import { ChallengeCardStatus, ChallengeDifficulty } from '../../database';
+import { setSageRole } from './setSageRole';
+
+const approveChallenge = async (
+  interaction: CommandInteraction | ButtonInteraction,
+  userId: string,
+) => {
+  const challengeCard = await challenges.loadChallengeCard(userId);
+  if (challengeCard) {
```

Comment:
There should be a check to ensure that the Challenge Card status is in the Approval state before we can update the challenge card to be approved or rejected. The same applies for the rejectChallenge.